### ISSUE TITLE
Table virtualization slows down the whole table, ultimate stuttering appears

### ISSUE THREAD

**Issue by raibikys** [rsuite/rsuite#2223]
### What version of rsuite are you using?

V5.30

### What version of React are you using?

17.0.2

### What version of TypeScript are you using (if any)?

4.3.2

### What browser are you using?

Chrome, Firefox

### Describe the Bug

Without virtualized table with larger size datasets (icons, div's, images, popovers) scrolls smoothly, but larger data affects reloading (infinite scrolling) behaviour, slows down all checkboxes or radio boxes above or under the table. 

**Main problem**: With virtualization "on" these problems are more or less gone, but then table breaks. Scrolling becomes very very lagging. User experience becomes bad overall. No need for 1000 records, 50 is enough to test this. After updating to V5.30 from V5.04 it became even worse.

### Expected Behavior

Smooth scrolling with virtualized on. Popovers or div's inside should affect user experience. I think it is with whole table problem itself.

### To Reproduce

Please check this fork:

https://codesandbox.io/s/silly-lake-4c91i

**Comment by raibikys** [comment]
@SevenOutman  @simonguo  any insights on this? :)

**Comment by simonguo** [comment]
hi @raibikys  We're still thinking about solutions.

Each mouse scroll will be re-rendered after virtualized is enabled. All visible areas of the table should be rendered with as little content as possible.
- Don't have too many Columns.
- The height of the Table should not be too high, e.g. from 900 to 500.

**Comment by raibikys** [comment]
Is 11 columns considered as too many? In real live example my table has no more than 700px height. I would use table without virtualization, but after i turn it off then checkboxes, inputs, etc. slows down arround the table. And also some weird infinite scrolling behaviour occurs.

**Comment by simonguo** [comment]
https://rsuite.github.io/rsuite-table/#8
This is an example containing 10000 rows of data, including checkbox, input and popover. Is there any interactive performance that is the problem on your computer?

**Comment by raibikys** [comment]
@simonguo no, this has nothing to do with computer. Tested on 3 different and powerful computers.

I think that pops out when more columns are added with popovers, icons, divs or other stuff. It has something to to with table perfomance itself.

**Comment by raibikys** [comment]
@simonguo I belive problem occurs when table height is set more than 300px. Just test it in my fork. Could the height be increased?

**Comment by simonguo** [comment]
https://codesandbox.io/s/eloquent-sid-8x2k1
In this example a React.memo is added to the GearCell and the Table height is changed to 500. Interaction performance will be improved.

**Comment by raibikys** [comment]
But with "virtualized" table can't scroll down.

**Comment by simonguo** [comment]
You upgrade rsuite to the latest 5.4.0 and try again.

**Comment by raibikys** [comment]
Hmm could not see any difference

**Comment by simonguo** [comment]
After using memo, the rendering time spent is about 4 ms. Before use, it was about 25 ms. From the data point of view, there is a significant improvement.
![image](https://user-images.githubusercontent.com/1203827/146904019-65f79a57-cfab-4bde-902f-eae00f1b4b7a.png)

**Comment by raibikys** [comment]
I see. Your forked example looks legit now. But in my own project I do not see any difference even when using React.memo along repetetive components. I'll try to provide better example.
